app/src/main/java/org/breezyweather/remoteviews/gadgetbridge/GadgetBridgeService.kt

```diff
@@ -52,17 +53,24 @@ object GadgetBridgeService {
     private const val GB_INTENT_PACKAGE_SMARTSPACER = "nodomain.pacjo.smartspacer.plugin.genericweather"
     private const val GB_INTENT_ACTION = "$GB_INTENT_PACKAGE.ACTION_GENERIC_WEATHER"
 
-    fun sendWeatherBroadcast(context: Context, location: Location) {
-        if (location.weather?.current == null) {
+    fun sendWeatherBroadcast(context: Context, locations: List<Location>) {
+        if (locations[0].weather?.current == null) {
             LogHelper.log(msg = "Not sending GadgetBridge data, current weather is null")
             return
         }
 
-        val weatherData = getWeatherData(context, location)
+        val nonNullLocations = locations.filter { it.weather?.current != null }
```

Comment:
I'm pretty sure this is what reduces the number of locations, can you retry your log with all locations even with null weather?

Looking at the code, I can see that only the first location is loaded with weather initially, which would explain why 2nd and 3rd locations are excluded by your code.

Reply:
Rebase on `main`, 37d2aa7be00cad365aa2e2c3d80ebc20b0e3d5f2 should help you here.

Data is sent to Gadgetbridge from main screen when:
- A location is added
- A location is updated
- A location is deleted
- Location order is swapped via drag & drop

Also in these cases that are probably useless:
- The view model of the main activity is initialized (I think I will keep that one, as it is useful for bringing back notifications/widgets that may have been killed)
- When you switch to view a different location
- When you open the app with a location as parameter, for example when you tap on a city from a widget, it will open the correct location

Reply:
This seems to have done the trick - I don't see it send the updates multiple times anymore.